### PR TITLE
Marker implementation

### PR DESCRIPTION
**Pull request by Focus**
Implemented markers #165. Along the way there was an issue with the way log parser was parsing #166 which is fixed.

### CODE REVIEW

**Comment by thomasjo** [comment]
Thanks for the contribution! :bow: 

I've started reviewing your changes, and overall it looks awesome. There are some style problems that will have to be addressed however. For some reason unknown to me right now, the CI linting is not behaving as expected and is currently producing false positives. In the meantime to uncover said style issues you can locally execute `standard`, e.g.
```bash
$ ./node_modules/.bin/standard lib/**/*.js
```

For your convenience, I dumped the errors into a gist:
https://gist.github.com/thomasjo/c7b77bb34fa0fdcd0d8f

---
*Side note: I'm looking into the false positives right now.*

**Comment by thomasjo** [comment]
The CI process now correctly fails on the style issues mentioned earlier.

---
To make it even easier to uncover these during local development, I'll look into providing some utility functions, as well as creating a contribution guide.

**Comment by Focus** [comment]
Thanks for the feedback. I fixed the issues with formatting. Let me know if it needs any more changes.

File: lib/error-marker.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    this.editor = editor
+    this.errors = errors
+    this.warnings = warnings
+    this.markers = []
+    this.mark()
+  }
+
+  mark () {
+    for (let error of this.errors) {
+      this.markRow(error.lineNumber - 1, 'line-red')
```
Comment (thomasjo) [inline lib/error-marker.js]: Change this class name from `line-red` to `latex-error`.

File: lib/error-marker.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    }
+    for (let warning of this.warnings) {
+      this.markRow(warning.lineNumber - 1, 'line-yellow')
```
Comment (thomasjo) [inline lib/error-marker.js]: Change this class name from `line-yellow` to `latex-warning`.

File: styles/markers.atom-text-editor.less [hunk 3]
```diff
@@ -0,0 +1,8 @@
+:host(:host) .gutter .line-number {
+  &.line-red {
+    background-color: red;
```
Comment (thomasjo) [inline styles/markers.atom-text-editor.less]: We should not be picking colors directly. Use `@background-color-error` as a basis instead. My initial suggestion here would be to mimic the `highlight-*` classes as defined in the Atom styleguide. E.g.

```less
@import "ui-variables";
// ...
  &.latex-error {
    background-color: fade(@background-color-error, 50%);
    color: @text-color-error;
  }
// ...
```

File: styles/markers.atom-text-editor.less [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+  }
+  &.line-yellow {
+    background-color: yellow;
```
Comment (thomasjo) [inline styles/markers.atom-text-editor.less]: Same as above, except use `@background-color-warning` and `@text-color-warning` instead.

**Comment by Focus** [comment]
Done

**Comment by thomasjo** [comment]
> Done

Excellent, but we also need to address the explicit color usage as I mentioned in https://github.com/thomasjo/atom-latex/pull/167#discussion_r54539563.

I can change it post-merge, but it would be great if you could refactor the style. Exactly what I want is this
```less
@import "ui-variables";

:host(:host) .gutter .line-number {
  &.latex-error {
    background-color: fade(@background-color-error, 50%);
    color: @text-color-error;
  }
  &.latex-warning {
    background-color: fade(@background-color-warning, 50%);
    color: @text-color-warning;
  }
}
```

Note that I haven't tested these styles, but they follow the official Atom styleguide, so I'm reasonably confident it will look OK.

**Comment by Focus** [comment]
Sorry I completely missed that. I implemented the changes you asked for. I also found a small bug where editor.getPath() would return undefined when one of the editors is a new page and fixed this as well.

Here are some screenshots under various themes:
atom-dark:
atom-dark
atom-light:
atom-light
one-dark:
one-dark
seti:
seti

**Comment by Focus** [comment]
I have given you access to my fork, so that you can push your own changes as well. I am more than happy to help with anything as well.

**Comment by thomasjo** [comment]
This looks really good! I can make some minor tweaks after merging if necessary.
Thanks again for this, such an awesome feature :sparkles:

**Comment by Focus** [comment]
Excellent. Thanks for feedback

**Pull request merged**